src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx

```diff
@@ -24,55 +39,79 @@ const LocationPreferences = () => {
         <div>
-          <InputFormError error={touched.country && errors.country}>
+          <InputFormError
+            error={
+              touched[LocationPreferencesFormKeys.COUNTRY] &&
+              errors[LocationPreferencesFormKeys.COUNTRY]
+            }
+          >
             <TextInputWithDropdown
               onChange={handleChange}
-              name="country"
+              name={LocationPreferencesFormKeys.COUNTRY}
               onBlur={handleBlur}
             />
           </InputFormError>
           <CheckboxInput
             onBlur={handleBlur}
-            id="openToRelocationCountry"
+            id={LocationPreferencesFormKeys.OPEN_TO_RELOCATION_COUNTRY}
             label="I’m open to residency relocation"
-            checked={values.openToRelocationCountry}
-            onChange={handleChange}
-            name="openToRelocationCountry"
-            dataTestId="openToRelocationCountry"
+            checked={
+              values[LocationPreferencesFormKeys.OPEN_TO_RELOCATION_COUNTRY]
+            }
+            onChange={() => {
+              handleToggleCheckbox(
+                '1',
+                LocationPreferencesFormKeys.OPEN_TO_RELOCATION_COUNTRY,
+              )
+            }}
+            name={LocationPreferencesFormKeys.OPEN_TO_RELOCATION_COUNTRY}
+            dataTestId={LocationPreferencesFormKeys.OPEN_TO_RELOCATION_COUNTRY}
           />
         </div>
         <div>
-          <InputFormError error={touched.city && errors.city}>
+          <InputFormError
+            error={
+              touched[LocationPreferencesFormKeys.CITY] &&
+              errors[LocationPreferencesFormKeys.CITY]
+            }
+          >
             <TextInput
               onBlur={handleBlur}
               label="City residency"
               placeholder="Start typing location"
-              value={values.city}
+              value={values[LocationPreferencesFormKeys.CITY]}
               onChange={handleChange}
               addImportantIcon={true}
-              name="city"
+              name={LocationPreferencesFormKeys.CITY}
               excludeDigits
               tooltipText="We use this information to match you with the best job opportunities."
-              dataTestId="city"
+              dataTestId={LocationPreferencesFormKeys.CITY}
               maxLength={40}
             />
           </InputFormError>
           <CheckboxInput
-            id="openToRelocationCity"
+            id={LocationPreferencesFormKeys.OPEN_TO_RELOCATION_CITY + 1}
             label="I’m open to city relocation"
-            checked={values.openToRelocationCity}
-            onChange={handleChange}
-            name="openToRelocationCity"
-            dataTestId="openToRelocationCity"
+            checked={
+              values[LocationPreferencesFormKeys.OPEN_TO_RELOCATION_CITY]
+            }
+            onChange={() => {
+              handleToggleCheckbox(
+                '1',
```

Comment:
issue, this function shouldn't use string, i've changed it to boolean. ('1' is a string not boolean required in schema)